File: use-cases/content-moderation/create-pattern.ts
/**
 * Create ImgGo Pattern for Content Moderation
 * Node.js/TypeScript version
 * Detect inappropriate content in images
 */

import axios from 'axios';
import * as fs from 'fs';
import * as dotenv from 'dotenv';

dotenv.config();

const API_KEY = process.env.IMGGO_API_KEY;
const BASE_URL = 'https://img-go.com/api';

interface PatternResponse {
  success: boolean;
  data: {
    id: string;
    name: string;
    format: string;
    endpoint_url: string;
  };
}

async function createModerationPattern(): Promise<string | null> {
  if (!API_KEY) {
    console.error('X Error: IMGGO_API_KEY not set');
    return null;
  }

  // Based on ImgGo API: format must be json, yaml, xml, csv, or text
  // All properties must be in required array
  const payload = {
    name: 'Content Moderation - JSON (Node.js)',
    instructions: 'Analyze image for inappropriate content. Detect: violence, adult content, hate symbols, weapons, drugs, self-harm. Return safety assessment and flagged categories.',
    format: 'json',
    json_schema: {
      type: 'object',
      properties: {
        is_safe: { type: 'boolean' },
        risk_level: { type: 'string' },
        flagged_categories: { type: 'string' },
        confidence: { type: 'number' }
      },
      required: ['is_safe', 'risk_level', 'flagged_categories', 'confidence'],
      additionalProperties: false
    }
  };

  console.log('='.repeat(60));
  console.log('CREATING CONTENT MODERATION PATTERN (Node.js)');
  console.log('='.repeat(60));
  console.log();

  try {
    const response = await axios.post<PatternResponse>(
      `${BASE_URL}/patterns`,
      payload,
      {
        headers: {
          'Authorization': `Bearer ${API_KEY}`,
          'Content-Type': 'application/json'
        }
      }
    );

    console.log(`Response Status: ${response.status}`);

    const patternId = response.data.data.id;

    console.log('V Pattern created successfully!');
    console.log();
    console.log(`Pattern ID: ${patternId}`);
    console.log();
    console.log('Add to .env:');
    console.log(`CONTENT_MODERATION_PATTERN_ID=${patternId}`);
    console.log();

    // Save pattern ID to file
    fs.writeFileSync('pattern_id.txt', patternId);
    console.log('V Saved to pattern_id.txt');

    return patternId;

  } catch (error) {
    if (axios.isAxiosError(error)) {
      console.error(`X Error: ${error.message}`);
      if (error.response) {
        console.error(`Response: ${JSON.stringify(error.response.data)}`);
      }
    } else {
      console.error(`X Error: ${error}`);
    }
    return null;
  }
}

// Run if called directly
createModerationPattern();

export { createModerationPattern };

File: use-cases/content-moderation/create-pattern.py
"""
Create ImgGo Pattern for Content Moderation
Detect inappropriate content in images
"""

import os
import requests

API_KEY = os.getenv("IMGGO_API_KEY")
BASE_URL = "https://img-go.com/api"

def create_moderation_pattern():
    if not API_KEY:
        print("X Error: IMGGO_API_KEY not set")
        return None

    payload = {
        "name": "Content Moderation - JSON",
        "instructions": "Analyze image for inappropriate content. Detect: violence, adult content, hate symbols, weapons, drugs, self-harm. Return confidence scores and flagged categories.",
        "response_format": "image_analysis",
        "schema": {
            "type": "object",
            "properties": {
                "is_safe": {"type": "boolean"},
                "risk_level": {"type": "string", "enum": ["safe", "warning", "unsafe"]},
                "flagged_categories": {"type": "array", "items": {"type": "string"}},
                "confidence_scores": {"type": "object"}
            }
        }
    }

    print("=" * 60)
    print("CREATING CONTENT MODERATION PATTERN")
    print("=" * 60)
    print()

    try:
        response = requests.post(
            f"{BASE_URL}/patterns",
            headers={"Authorization": f"Bearer {API_KEY}", "Content-Type": "application/json"},
            json=payload
        )
        response.raise_for_status()

        pattern_id = response.json().get("data", {}).get("id")
        print(f"V Pattern created successfully!\n\nPattern ID: {pattern_id}\n")
        print(f"Add to .env:\nCONTENT_MODERATION_PATTERN_ID={pattern_id}\n")

        with open("pattern_id.txt", "w") as f:
            f.write(pattern_id)
        print("V Saved to pattern_id.txt")
        return pattern_id

    except Exception as e:
        print(f"X Error: {e}")
        return None

if __name__ == "__main__":
    create_moderation_pattern()
